extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -336,25 +337,29 @@ private void handleOpenTracing(BuildProducer<ReflectiveClassBuildItem> reflectiv
                 "io.opentracing.contrib.kafka.TracingConsumerInterceptor"));
     }
 
-    private void handleStrimziOAuth(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+    private void handleStrimziOAuth(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         if (!QuarkusClassLoader.isClassPresentAtRuntime("io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler")) {
             return;
         }
 
         reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
                 "io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler"));
 
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
-                "org.keycloak.jose.jws.JWSHeader",
-                "org.keycloak.representations.AccessToken",
-                "org.keycloak.representations.AccessToken$Access",
-                "org.keycloak.representations.AccessTokenResponse",
-                "org.keycloak.representations.IDToken",
-                "org.keycloak.representations.JsonWebToken",
-                "org.keycloak.jose.jwk.JSONWebKeySet",
-                "org.keycloak.jose.jwk.JWK",
-                "org.keycloak.json.StringOrArrayDeserializer",
-                "org.keycloak.json.StringListMapDeserializer"));
+        if (curateOutcomeBuildItem.getApplicationModel().getDependencies().stream().anyMatch(
+                x -> x.getGroupId().equals("org.keycloak") && x.getArtifactId().equals("keycloak-core"))) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
```

Comment:
What did we decide about this pattern? IIRC, there was a discussion in another PR

Reply:
It looks like there is no better way at the moment (see https://github.com/quarkusio/quarkus/pull/29886#discussion_r1054703116)

Reply:
Understood, thanks